mail_multi_website_email/models/res_partner_hook_get_signup_url_for_action.py

```diff
@@ -0,0 +1,86 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 Quartile Limited
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+import werkzeug
+from urlparse import urljoin
+from odoo import models, fields, api, http
+from odoo.addons.auth_signup.models.res_partner import ResPartner
+
+
+# Monkey Patching
+# Overwrite the original _get_signup_url_for_action in auth_signup
+# i.e. https://github.com/odoo/odoo/blob/10.0/addons/auth_signup/models
+# /res_partner.py#L49-L94
+@api.multi
+def _get_signup_url_for_action(self, action=None, view_type=None,
+                               menu_id=None, res_id=None, model=None):
+    """ generate a signup url for the given partner ids and action, possibly overriding
+        the url state components (menu_id, id, view_type) """
+
+    res = dict.fromkeys(self.ids, False)
+    """QTL MOD - The base url could be different among users"""
+    #base_url = self.env['ir.config_parameter'].get_param('web.base.url')
+    for partner in self:
+
+        """QTL MOD - Search if the user account has the multi-website
+        settings and set up the base url with correct domain
+        """
+        base_url = http.request.env[
+            'ir.config_parameter'].get_param('web.base.url')
+        domain = [
+            ('partner_id', '=', partner.id)
```

Comment:
No need to include `(active, '=', True)`?  I guess it is technically possible to link multiple users to a partner although it shouldn't happen under normal circumstances.